sequencer.c

```diff
@@ -1507,6 +1528,13 @@ static int try_to_commit(struct repository *r,
 
 	reset_ident_date();
 
+	if (opts->ignore_date) {
+		ignore_author_date(&author);
+		if (flags & AMEND_MSG) {
+			free(amend_author);
+			amend_author = (char *)author;
```

Comment:
It seems to me that `amend_author` is only used to free `author` when we get it from `get_author`, instead of from the parameter that's passed in (and should not be free'd.

So it seems to me that we should always assign `author` to `amend_author` here to avoid ano memory leaks.  At that point `amend_author` is no longer the right name, so maybe we should rename it to `author_to_free` or something in a preparatory patch?

Reply:
Yes, it is indeed used to do the aforementioned. Let's account all your suggested changes in some commits.